src/plugins/discover/public/application/main/hooks/use_text_based_query_language.ts

```diff
@@ -120,14 +121,16 @@ export function useTextBasedQueryLanguage({
               prev.current.query = query[language];
               prev.current.columns = nextColumns;
             }
-            const nextState = {
-              ...(addDataViewToState && { index: dataViewObj.id }),
-              ...((addColumnsToState || queryChanged) && { columns: nextColumns }),
```

Comment:
It looks like we used to always update the columns, since I think `queryChanged` would always be true here. Does changing this impact anything?

Reply:
you're right, it looks like that. but this way it should be better, since we just update the state when needed, when previous columns !== next columns